v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -29,14 +29,16 @@ type ARMClientCache struct {
 	kubeClient         kubeclient.Client
 	httpClient         *http.Client
 	armMetrics         *metrics.ARMClientMetrics
+	userAgent          string
 }
 
 func NewARMClientCache(
 	credentialProvider identity.CredentialProvider,
 	kubeClient kubeclient.Client,
 	configuration cloud.Configuration,
 	httpClient *http.Client,
-	armMetrics *metrics.ARMClientMetrics) *ARMClientCache {
+	armMetrics *metrics.ARMClientMetrics,
+	userAgent string) *ARMClientCache {
```

Comment:
Instead of plumbing the UA through here, what if there was a method in the `genericarmclient` package that could be called to _add_ a string to the UA for this process?

The code in `setup.go` would then be

``` 
genericarmclient.AddToUserAgent(cfg.UserAgentSuffix)
```

The variable `DefaultUserAgent` woudl revert to package internal and everything else would continue as before.

Reply:
Good call, that's a lot less involved.